main/mio.c

```diff
@@ -999,8 +1004,10 @@ char *mio_gets (MIO *mio, char *s, size_t size)
 				s[i] = 0;
 				rv = s;
 			}
-			if (mio->impl.mem.pos >= mio->impl.mem.size)
+			if (!newline && pos >= buf_size)
```

Comment:
Anyway: please do not change MIO memory implementation if it doesn't match the FILE implementation.